data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -314,10 +389,13 @@ class ApiServerClient {
           patchBody.status = statusJson;
         }
         logger.info(`Updating - Resource ${opts.resourceId} with body - ${JSON.stringify(patchBody)}`);
+        const namespaceId = this.getNamespaceId(opts.resourceId);
+        // Create Namespace if not default
         return Promise.try(() => this.init())
+          .then(() => this.createNamespace(namespaceId))
```

Comment:
during create namespace would have been created already. Why is this being created if not exists during update? Rather shouldnt this throw an exception?

Reply:
Agree. @amitmalav can you pls take care. I would say lets remove it altogether and let the error from apiserver flow in.

Reply:
Removed from update methods.